pages/en/lb4/Decorator.md

```diff
@@ -0,0 +1,209 @@
+values!) in other areas of your code.
+
+```ts
+// application.ts
+import {Application} from '@loopback/core';
+import 'fs-extra';
+class MyApp extends Application {
+  constructor() {
+    super();
+    const app = this;
+    const widgetConf = JSON.parse(fs.readSync('./widget-config.json'));
+    app.bind('config.widget').to(widgetConf);
+  }
+}
+```
+Now that we've bound the 'config.widget' key to our configuration object, we can
+inject it in our WidgetController:
+
+```ts
+// widget-controller.ts
+import {widgetSpec} from '../apispec';
+@api(widgetSpec)
+class WidgetController {
+  @inject('config.widget')
+  protected config: JSON; // This will be resolved at runtime!
+
+  // etc...
+}
+```
+
+For more information, see the [Dependency Injection](Dependency-Injection.htm) section under [LoopBack Core Concepts](Concepts.htm)
+
+## Authentication Decorator
+
+- @authenticate
+
+  Mark a controller method as requiring authenticated user, takes a strategy name 
+  as a parameter. An example is using 'BasicStrategy' to authenticate user in function
+  `whoAmI`:
+
+  ```ts
+  // my-controller.ts
+  import { authenticate } from '@loopback/authentication';
+  import { inject } from '@loopback/context';
+
+  class MyController {
+    constructor(
+      @inject(BindingKeys.Authentication.CURRENT_USER)
+      private user: UserProfile,
+    ) {}
+
+    @authenticate('BasicStrategy')
+    async whoAmI() : Promise<string> {
+      return this.user.id;
+    }
+  }
+  ```
+
+## Repository Decorators
+
+### Model decorators
+
+- @model
+
+  Define a model in a repository, see [define-models](http://loopback.io/doc/en/lb4/Repositories.html#define-models)
+
+- @property
+
+  Define a property in a model, see [define-models](http://loopback.io/doc/en/lb4/Repositories.html#define-models)
+
+### Relation decorators
+
+  *This feature has not yet been released in alpha form. Documentation will be* 
+  *added here as this feature progresses.*
+
+- @relation
+
+  Register a general relation
+
+- @belongsTo @hasOne @hasMany @embedsOne @embedsMany @referencesOne @referencesMany
+
+  Register a specific relation
+
+### Repository decorators
+
+- @repository
+
+  A repository represents a collection of data and the means to access that data, 
+  the decorator either injects an existing repository or creates a repository 
+  from a model and a datasource.
+
+  The injection example can be found in [Repository#controller-configuration](Repositories.html#controller-configuration)
+
+  ```ts
+  // my-controller.ts
+  import { Todo } from '{path_of_Todo_mode}.ts';
+  import { juggler, DataSourceConstructor } from '@loopback/repository';
+
+  const datasource = new DataSourceConstructor({
+    connector: 'mysql',
+    host: 'localhost',
+    port: 3306,
+    database: 'test',
+    password: 'pass',
+    user: 'root',
+  });
+
+  export class TodoController {
+    @repository(Todo, datasource)
+    repository: EntityCrudRepository<Todo, number>;
```

Comment:
While this is a valid usage, we are advocating for a slightly different usage in Thinking in LoopBack - see [Define Product model, repository, and data source](http://loopback.io/doc/en/lb4/Thinking-in-LoopBack.html#define-product-model-repository-and-data-source) and [Decouple Controller from Repository](http://loopback.io/doc/en/lb4/Thinking-in-LoopBack.html#decouple-controller-from-repository).

I'd like us to follow the approach described in Thinking in LoopBack in the examples shown on this page too.